tests/algorithm_validation_tests/helpers.py

```diff
@@ -23,6 +22,14 @@ def algorithm_request(algorithm: str, input: dict):
     return response
 
 
+def algorithm_request_no_parse(algorithm: str, input: dict):
```

Comment:
The functions `algorithm_request` and `algorithm_request_no_parse` are redundant. There should be only one function, `algorithm_request` that does exactly what is says, i.e. make a post request and return the result. Verification of the expected status code and/or validation of the response contents is the responsibility of the individual tests.

Reply:
Why duplicate the check for the status and the parse in every test? 

For successful tests it's always the same.

Reply:
If the intention of the tests is to verify that the status code is something specific, then the check will appear in every test. DRY principle is not relevant here.

On the other hand, if the tests only need to assert that the result is correct, and the code is assumed to be 200, then DRY principle is applicable and parsing can be abstracted to a function. But the obvious way to decompose is 1) `algorithm_request`: makes algorithm request and returns response and 2) `parse_response`: parses `json` response if status code is 200. 

Not `algorithm_request` and `algorithm_request_no_parse`.

Reply:
I think that the intention of most algorithm validation tests is to verify the result values, not the status or the parsing.

So each test will use something like `result = parse_response(algorithm_request(algorithm_name, input))`

If we can't find a name that will convey the meaning of both `algorithm_request` and `parse_response`, to avoid using both each time it looks to me as well like the best approach.

Reply:
You seriously think that `algorithm_request` and `algorithm_request_no_parse` is a better decomposition than `algorithm_request` and `parse_response`, because **one line of code is too much repetition**?? 

You do realize that calling one function that does two things 
```
algorithm_request_and_parse_response(algorithm_name, input)
``` 
has almost the same character count as two well defined functions 
```
parse_response(algorithm_request(algorithm_name, input))
```
?

Reply:
I never said to stick with the name `algorithm_request_no_parse`, I said that we can stick with `parse_response(algorithm_request(algorithm_name, input))` or find a better name that will include both calls.

Before this PR `algorithm_request` was doing both the request and the parsing, without the extra `_and_parse_response` in the name.

At this stage I don't think it's very important what way we choose. 

The important think IMO is not to duplicate the status check and the parsing, as it was initially suggested and IF possible keep 2 separate methods with parsing and without. 

Is that clear?

Reply:
Chat GPT to the rescue:


Here are some names without the word "and":

    execute_algorithm_with_parsed_result: This name suggests that the function executes the algorithm and returns the parsed result.

    run_algorithm_with_parsed_output: This name conveys that the function runs the algorithm and returns the parsed output.

    fetch_parsed_algorithm_response: This name implies that the function fetches the algorithm response and returns the parsed result.

It's important to choose a name that is easy to understand and accurately describes what the function does. These names convey that the function is executing the algorithm and returning a parsed result, without explicitly using the word "and".


😆